dbt/adapters/athena/impl.py

```diff
@@ -640,6 +652,10 @@ def get_columns_in_relation(self, relation: AthenaRelation) -> List[Column]:
             glue_client = client.session.client("glue", region_name=client.region_name, config=get_boto3_config())
 
         table = glue_client.get_table(DatabaseName=relation.schema, Name=relation.identifier)["Table"]
-        return [
-            Column(c["Name"], c["Type"]) for c in table["StorageDescriptor"]["Columns"] + table.get("PartitionKeys", [])
-        ]
+
+        columns = [c for c in table["StorageDescriptor"]["Columns"] if self._is_current_column(c) is True]
```

Comment:
```suggestion
        columns = [c for c in table["StorageDescriptor"]["Columns"] if self._is_current_column(c)]
```
seems to redundant check here

Reply:
indeed, good point.